cmd/otel-allocator/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator
 
-go 1.20
+go 1.21
```

Comment:
Is it necessary to increase the versions here?  This will prevent building with a Go toolchain less than 1.21.  I don't think we're taking any dependencies on 1.21 here.

Reply:
+1